faucet/src/http.rs

```diff
@@ -147,30 +138,41 @@ fn has_request_expired(
 async fn ensure_funding_allowed(
     parachain_rpc: &InterBtcParachain,
     account_id: AccountId,
-    currency_id: CurrencyId,
+    allowance_config: AllowanceConfig,
     last_request_json: Option<Json<FaucetRequest>>,
     account_type: FundingRequestAccountType,
 ) -> Result<(), Error> {
-    let free_balance = parachain_rpc
-        .get_free_balance_for_id(account_id.clone(), currency_id)
-        .await?;
-    let reserved_balance = parachain_rpc
-        .get_reserved_balance_for_id(account_id.clone(), currency_id)
-        .await?;
-    if free_balance + reserved_balance > MAX_FUNDABLE_CLIENT_BALANCE * currency_id.inner()?.one() {
-        log::warn!(
-            "User {} has enough funds: {:?}",
-            account_id,
-            free_balance + reserved_balance
-        );
-        return Err(Error::AccountBalanceExceedsMaximum);
+    // Could get the currencies from either `user_allowances` or `vault_allowances`, assuming the same ones are used.
+    let currency_ids: Result<Vec<_>, _> = allowance_config
+        .user_allowances
+        .iter()
+        .map(|x| CurrencyId::try_from_symbol(x.symbol.clone()))
+        .collect();
+    for currency_id in currency_ids?.iter() {
+        let free_balance = parachain_rpc
+            .get_free_balance_for_id(account_id.clone(), *currency_id)
+            .await?;
+        let reserved_balance = parachain_rpc
+            .get_reserved_balance_for_id(account_id.clone(), *currency_id)
+            .await?;
+
+        let one = |currency: CurrencyId| Ok::<_, Error>(10u128.pow(currency.decimals()? as u32));
+        if free_balance + reserved_balance > allowance_config.max_fundable_client_balance * one(*currency_id)? {
```

Comment:
I'm wondering if `max_fundable_client_balance` should be a map? Not sure if it makes to have the same max for every currency

Reply:
We could but since this faucet is only for testnet I'm not sure how much it's worth focusing on every detail